tiles-generation/dockerAssets/entrypoint.sh

```diff
@@ -0,0 +1,143 @@
+    if [ "${DOWNLOAD_PLANET:-}" == "enabled" ] || [ "${DOWNLOAD_PLANET:-}" == "1" ]; then
+        latestPlanetPBFFilePath=$(aws s3 ls s3://osm-pds/ --recursive --no-sign-request | sort | awk '{print $4}' | grep 'planet-' | grep '.*\.pbf' | grep -v '.md5' | grep -v '.torrent' | tail -1)
+        aws s3 cp s3://osm-pds/${latestPlanetPBFFilePath} . --no-sign-request
+        latestPlanetPBFFileName=$(basename ${latestPlanetPBFFilePath})
+        mv ./${latestPlanetPBFFileName} /data.osm.pbf
+    fi
+
+    #Download extacts from OpenStreetMap from https://download.geofabrik.de/
+    if [ -n "${DOWNLOAD_PBF:-}" ]; then
+        echo "INFO: Download PBF file: $DOWNLOAD_PBF"
+        wget ${WGET_ARGS:-} "$DOWNLOAD_PBF" -O /data.osm.pbf
+    fi
+
+    if [ -n "${DOWNLOAD_POLY:-}" ]; then
+        echo "INFO: Download PBF-POLY file: $DOWNLOAD_POLY"
+        wget ${WGET_ARGS:-} "$DOWNLOAD_POLY" -O /data.poly
+    fi
+
+    # copy polygon file if available
+    if [ -f /data.poly ]; then
+        sudo -u renderer cp /data.poly /var/lib/mod_tile/data.poly
+    fi
+
+    # Import data
+    sudo -u renderer osm2pgsql -d gis --create --slim -G --hstore --tag-transform-script /home/renderer/src/openstreetmap-carto/openstreetmap-carto.lua --number-processes ${THREADS:-4} -S /home/renderer/src/openstreetmap-carto/openstreetmap-carto.style /data.osm.pbf ${OSM2PGSQL_EXTRA_ARGS:-}
+
+    # Create indexes
+    sudo -u postgres psql -d gis -f /home/renderer/src/openstreetmap-carto/indexes.sql
+
+    #Import external data
+    sudo chown -R renderer: /home/renderer/src
+    RETRY_NUM=5
+    RETRY_EVERY=10
+    NUM=$RETRY_NUM
+    until  sudo -E -u renderer python3 /home/renderer/src/openstreetmap-carto/scripts/get-external-data.py -c /home/renderer/src/openstreetmap-carto/external-data.yml -D /home/renderer/src/openstreetmap-carto/data
+    do
+    1>&2 echo failure import external data from openstreetmap-carto/scripts/get-external-data.py... retrying $NUM more times
+    sleep $RETRY_EVERY
+    ((NUM--))
+    if [ $NUM -eq 0 ]
+    then
+        1>&2 echo command was not successful after $RETRY_NUM tries
+        exit 1
+    fi
+    done
+
+    service postgresql stop
+}
+
+function servemap() {
+    # Clean /tmp
+    rm -rf /tmp/*
+
+    # Fix postgres data privileges
+    chown postgres:postgres /var/lib/postgresql -R
+
+    # Privileges for writing tiles image
+    chmod 777 /var/lib/mod_tile -R
+
+    # Initialize PostgreSQL and Apache
+    createPostgresConfig
+    service postgresql start
+    service apache2 restart
+    setPostgresPassword
+
+    # Configure renderd threads
+    sed -i -E "s/num_threads=[0-9]+/num_threads=${THREADS:-4}/g" /usr/local/etc/renderd.conf
+
+    # Generate tile scripts
+    sudo -E -u postgres /var/lib/postgresql/src/generate_tiles.py
+
+    # Upload to S3 Bucket
+    currentTime=$(date "+%Y%m%d%H%M%S")
+    aws s3 cp /var/lib/mod_tile s3://${TILE_S3_BUCKET}/tiles_${currentTime} --recursive
+
```

Comment:
Could we reduce the capacity of ASG at end of Docker script?

Reply:
That's a good idea, it will require permission role and arn to execute this kind of command to a specify ASG, let me dive deep to see if it can be workable.

Reply:
I updated the [ASG strategy](https://github.com/opensearch-project/maps/blob/64bb9ccef8f35563886dafbdac3a0b3cf3b12d2a/tiles-generation/cdk/lib/tile-generation-stack.ts#L54-L56) to make ec2 instance automatically closed after executing the task by aws cli. Base on the new strategy, we will don't need to manage the instance. More reading on ECS Cluster Auto Scaling [here](https://aws.amazon.com/blogs/containers/deep-dive-on-amazon-ecs-cluster-auto-scaling/)